src/ModalManager.test.js

```diff
@@ -0,0 +1,495 @@
+    <button type="button" onClick={onCancel}>{cancelButton}</button>
+  </div>
+)));
+jest.mock('./components/ClaimedReturnedModal', () => jest.fn(({ onCancel, onConfirm }) => (
+  <div data-testid="claimedReturnedModal">
+    <button type="button" onClick={onConfirm}>{confirmButton}</button>
+    <button type="button" onClick={onCancel}>{cancelButton}</button>
+  </div>
+)));
+jest.mock('./components/MultipieceModal', () => jest.fn(({ onConfirm, onClose }) => (
+  <div data-testid="multipieceModal">
+    <button type="button" onClick={onConfirm}>{confirmButton}</button>
+    <button type="button" onClick={onClose}>{cancelButton}</button>
+  </div>
+)));
+
+describe('ModalManager', () => {
+  const labelIds = {
+    checkinNotesHeading: 'ui-checkin.checkinNotes.heading',
+    checkinNoteModalHeading: 'ui-checkin.checkinNoteModal.heading',
+    close: 'ui-checkin.close',
+    multipieceModalCancel: 'ui-checkin.multipieceModal.cancel',
+    multipieceModalConfirm: 'ui-checkin.multipieceModal.confirm',
+    checkinNotesMessage: 'ui-checkin.checkinNotes.message',
+    checkinNoteModalMessage: 'ui-checkin.checkinNoteModal.message',
+    date: 'ui-checkin.date',
+    note: 'ui-checkin.note',
+    source: 'ui-checkin.source',
+    confirmModalHeading: 'ui-checkin.confirmModal.heading',
+    confirmModalMessage: 'ui-checkin.confirmModal.message',
+    confirmModalCancel: 'ui-checkin.confirmModal.cancel',
+    confirmModalConfirm: 'ui-checkin.confirmModal.confirm',
+    confirmButton,
+    cancelButton,
+  };
+  const testIds = {
+    checkinNoteModal: 'checkinNoteModal',
+    claimedReturnedModal: 'claimedReturnedModal',
+    multipieceModal: 'multipieceModal',
+    confirmationModal: 'confirmationModal',
+  };
+  const notesWithCheckinType = [
+    {
+      noteType: statuses.CHECK_IN,
+      date: 'dateA',
+      desc: 'descA',
+    }, {
+      noteType: statuses.CHECK_IN,
+      date: 'dateB',
+      desc: 'descA',
+    }, {
+      noteType: statuses.CHECK_IN,
+      date: 'dateA',
+      desc: 'descB',
+    },
+  ];
+  const noteWithoutCheckinType = {
+    noteType: statuses.UNKNOWN,
+    date: 'dateB',
+    desc: 'descB',
+  };
+  const checkedinItem = {
+    title: 'title',
+    barcode: 'barcode',
+    discoverySuppress: true,
+    status: {
+      name: statuses.CLAIMED_RETURNED,
+    },
+    materialType: {
+      name: 'book',
+    },
+    circulationNotes: [...notesWithCheckinType, noteWithoutCheckinType],
+  };
+  const claimedReturnedHandler = jest.fn();
+  const onDone = jest.fn();
+  const onCancel = jest.fn();
+  const defaultProps = {
+    checkedinItem,
+    claimedReturnedHandler,
+    onDone,
+    onCancel,
+  };
+
+  describe('when "checkinNotesMode" is true', () => {
+    beforeEach(() => {
+      render(
+        <ModalManager
+          {...defaultProps}
+          checkinNotesMode
+        />
+      );
+    });
+
+    afterEach(() => {
+      jest.clearAllMocks();
```

Comment:
looks like it should be in the first describe?

Reply:
You are right, thanks!